### PR TITLE
Attempt at replacing Helper.log calls with corresponding UI.* ones

### PR DESCRIPTION
**Pull request by lacostej**
Note that some calls were a bit strange, so I had to take a decision. For example:

```
  # fatal -> error - is this what we want ?
  expect_fixed(
    'Helper.log.fatal "HI".red',
    'UI.error "HI"'
  )
  # info yellow => important - is this what we want ?
  expect_fixed(
    'Helper.log.info "HI".yellow',
    'UI.important "HI"'
  )
  # we don't do anything about magenta
  expect_fixed(
    'Helper.log.info "\'#{key}\' enabled.".magenta if value',
    'UI.message "\'#{key}\' enabled.".magenta if value'
  )
  # we don't do anything about yellow for debug - is that what we want ?
  expect_fixed(
    'Helper.log.debug "yo".yellow',
    'UI.verbose "yo"'
  )
```

I can provide the script that do the job if needed.

### CODE REVIEW

**Comment by lacostej** [comment]
image

Doesn't sound like an issue with my PR.

**Comment by KrauseFx** [comment]
This should be fixed with https://github.com/fastlane/fastlane/pull/3688

**Comment by KrauseFx** [comment]
This looks great, thanks for the PR @lacostej. Due to https://github.com/fastlane/fastlane/pull/3660 there are now some conflicts. 
What's even more important than migrating the `Helper.log.` calls, is to get rid of all `raise` blocks. 

```
# fastlane "crash" because of a user error
# everything that is caused by the user and is not unexpected
UI.user_error!("You don't have a project in the current directory")

# an actual crash when something unexpected happened
UI.crash!("Network timeout")
```


The docs are available here https://github.com/fastlane/fastlane/tree/master/countdown#interacting-with-the-user

**Comment by KrauseFx** [comment]
The `raise` blocks can not really be automated, so I'm more than happy to take over the second part. Could you re-run the script on the latest master to resolve the conflicts? :+1:

**Comment by lacostej** [comment]
note that I've just re-ran my custom update script.

I can try making one for user_error as well, given some heuristics, like in this one.

**Comment by KrauseFx** [comment]
I just saw that you separated this PR in 2 commits. Could the parser use
```ruby
UI.message("something")
```
instead of
```ruby
UI.message "something"
```

I really like having `(` around method calls, since things get confusing really fast otherwise

```ruby
UI.message "something" unless Helper.ci?
```

**Comment by lacostej** [comment]
@KrauseFx do you want me to convert all code, when possible, do use () ?

**Comment by lacostej** [comment]
@KrauseFx PR updated.

**Comment by KrauseFx** [comment]
This looks great, thank you! Reviewing the source code now :+1:

**Comment by KrauseFx** [comment]
:+1: from me, I'd like a second pair of :eyes: for this, just to be sure

File: fastlane/lib/fastlane/actions/artifactory.rb [hunk 1]
```diff
@@ -15,15 +15,15 @@ def self.run(params)
...
-          Helper.log.info "Size: #{upload.size}"
-          Helper.log.info "SHA1: #{upload.sha1}"
+          UI.message("Uploaded Artifact:")
+          UI.message("Repo: #{upload.repo}")
+          UI.message("URI: #{upload.uri}")
+          UI.message("Size: #{upload.size}")
+          UI.message("SHA1: #{upload.sha1}")
         else
-          Helper.log.info "File not found: '#{file_path}'"
+          UI.message("File not found: '#{file_path}'")
```
Comment (jcampbell05) [inline fastlane/lib/fastlane/actions/artifactory.rb]: Should this be an error or a warning ? I don't think it should be a standard message.

File: fastlane/lib/fastlane/actions/bundle_install.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def self.run(params)
 
           return sh(cmd.join(' '))
         else
-          Helper.log.info "No Gemfile found"
+          UI.message("No Gemfile found")
```
Comment (jcampbell05) [inline fastlane/lib/fastlane/actions/bundle_install.rb]: Should this be an error or a warning ? I don't think it should be a standard message.

File: fastlane/lib/fastlane/actions/bundle_install.rb [hunk 3]
```diff
@@ -40,7 +40,7 @@ def self.gemfile_exists?(params)
         possible_gemfiles.each do |gemfile|
           gemfile = File.absolute_path(gemfile)
           return true if File.exist?(gemfile)
-          Helper.log.info "Gemfile not found at: '#{gemfile}'"
+          UI.message("Gemfile not found at: '#{gemfile}'")
```
Comment (jcampbell05) [inline fastlane/lib/fastlane/actions/bundle_install.rb]: Should this be an error or a warning ? I don't think it should be a standard message.

File: fastlane/lib/fastlane/actions/changelog_from_git_commits.rb [hunk 4]
```diff
@@ -10,11 +10,11 @@ def self.run(params)
...
           from = Actions.last_git_tag_name(params[:match_lightweight_tag], params[:tag_match_pattern])
-          Helper.log.debug "Found the last Git tag: #{from}"
+          UI.verbose("Found the last Git tag: #{from}")
           to = 'HEAD'
         end
 
-        Helper.log.info "Collecting Git commits between #{from} and #{to}".green
+        UI.success("Collecting Git commits between #{from} and #{to}")
 
         # Normally it is not good practice to take arbitrary input and convert it to a symbol
```
Comment (jcampbell05) [inline fastlane/lib/fastlane/actions/changelog_from_git_commits.rb]: In a future PR we should deprecate the `: include_merges` option for the `: merge_commit_filtering` option.

**Comment by jcampbell05** [comment]
@lacostej Overall looks great just left a few smaller comments :)

File: fastlane/lib/fastlane/actions/artifactory.rb (hunk 1 above)
Comment (lacostej) [inline fastlane/lib/fastlane/actions/artifactory.rb]: Note: I wrote a script to rewrite all these calls, and it's not taking into account the original "errors" in API usage. If fine for you, I would prefer to manually correct those errors in a separate PR as they are semantically different (remove usage of deprecated API vs fix logging level).

Comment (lacostej) [inline fastlane/lib/fastlane/actions/artifactory.rb]: Or at least in a separate commit :)

Comment (jcampbell05) [inline fastlane/lib/fastlane/actions/artifactory.rb]: Okay separate PR makes sense :)

**Comment by jcampbell05** [comment]
@KrauseFx looks good to me :) @lacostej will fix the things I brought up in a separate PR :)

File: fastlane/lib/fastlane/actions/ssh.rb [hunk 5]
```diff
@@ -48,17 +48,17 @@ def self.run(params)
...
             raise "SSH Command failed" if return_value[:exit_code] > 0
 
             stderr << return_value[:stderr]
             stdout << return_value[:stdout]
           end
         end
-        Helper.log.info "Succesfully executed #{params[:commands].count} commands on host: #{params[:host]}"
-        Helper.log.info "\n########### \n #{stdout} \n###############".magenta if params[:log]
+        UI.message("Succesfully executed #{params[:commands].count} commands on host: #{params[:host]}")
+        UI.message("\n########### \n #{stdout} \n###############".magenta) if params[:log]
```
Comment (i2amsam) [inline fastlane/lib/fastlane/actions/ssh.rb]: This looks like the right choice to me :+1:

**Comment by i2amsam** [comment]
This looks good to me too!  :100:

File: fastlane/lib/fastlane/actions/artifactory.rb (hunk 1 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/artifactory.rb]: Yeah :+1:

**Comment by KrauseFx** [comment]
Thanks everyone. @jcampbell05, do you want to submit a PR with the warning/error changes? :+1:

**Comment by KrauseFx** [comment]
Thanks @lacostej for the awesome PR, this saved us a ton of time :rocket:

**Comment by jcampbell05** [comment]
@krauseFX sure :)

**Pull request merged**